BraintreeCore/src/main/java/com/braintreepayments/api/core/GetReturnLinkTypeUseCase.kt

```diff
@@ -0,0 +1,32 @@
+package com.braintreepayments.api.core
+
+import android.content.Intent
+import android.content.pm.PackageManager
+import androidx.annotation.RestrictTo
+import com.braintreepayments.api.core.GetReturnLinkTypeUseCase.ReturnLinkType
+
+/**
+ * Use case that returns which link type should be used for navigating from App Switch / CCT back into the merchant app.
+ *
+ * If a user unchecks the "Open supported links" checkbox in the Android OS settings for the merchant's app. If this
+ * setting is unchecked, this use case will return [ReturnLinkType.DEEP_LINK], otherwise [ReturnLinkType.APP_LINK]
+ * will be returned.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class GetReturnLinkTypeUseCase(private val merchantRepository: MerchantRepository) {
```

Comment:
Thoughts on adding a domain layer / use case pattern to the SDK? I wasn't able to find an existing pattern to share business logic outside of creating additional Client classes.

Reply:
Can you expand on what you envision for adding a domain layer?

Reply:
Yep! I see it as a layer that encapsulates business logic that can be reused across the SDK. It's also the layer between the UI and the data layer, where it reads data from the repository and applies some sort of business logic.

https://developer.android.com/topic/architecture/domain-layer

Reply:
I'm okay with this pattern. At some point we should consider documenting the Android and iOS patterns internally since they have diverged quite a bit in recent major versions. Unless this is more of a best practice in Android, in which case probably fine to leave undocumented and just slowly migrate to each languages best practices!

Reply:
I'd say MVVM is the architecture pattern that Google is pushing since they have entire doc sections on the different layers of MVVM. That said, it wouldn't hurt to have a callout somewhere in the repo of the architecture patterns the SDK is using.

Reply:
Pretty widely used pattern in android. AFAIK, folks on NXO iOS also made use of this.